### PR TITLE
Create RESTEasy Jackson codestart

### PR DESCRIPTION
**Pull request by jantosi**
This is provided instead of https://github.com/quarkusio/quarkus/pull/12341 - rebased commits.

### CODE REVIEW

**Comment by ia3andy** [comment]
@jantosi you don't need to create a new PR when rebasing :), rebase locally, then use `git push jantosi --force-with-lease`

File: devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/examples/resteasy-jackson-example/java/src/test/java/org/acme/rest/json/NativeFruitResourceIT.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package org.acme.rest.json;
```
Comment (ia3andy) [inline devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/examples/resteasy-jackson-example/java/src/test/java/org/acme/rest/json/NativeFruitResourceIT.java:1]: You need to move the native test to `src/native-test/java/org/acme/rest/json/NativeFruitResourceIT.java`, this is a `gradle` specificity (for maven the file will be automatically moved to the test directory), maybe we need to add this to the doc..

File: devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/examples/resteasy-jackson-example/kotlin/src/test/kotlin/org/acme/rest/json/NativeFruitResourceIT.kt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package org.acme.rest.json
```
Comment (ia3andy) [inline devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/examples/resteasy-jackson-example/kotlin/src/test/kotlin/org/acme/rest/json/NativeFruitResourceIT.kt:1]: Same as the other, you need to move the native test.

**Review by ia3andy — CHANGES_REQUESTED** [review]
CI is failing on gradle, I added comments on the reason..

File: independent-projects/tools/codestarts/README.adoc [hunk 3]
```diff
@@ -8,6 +8,8 @@ This guide explains how to create and configure a Quarkus Codestart for an exten
 Codestarts provide a personalized Quarkus getting started experience and really show the Quarkus breadth.
 A Quarkus extension are able to provide one or more well defined codestarts which will contain the necessary resources and code examples required to get started using that particular extension.
 
+See the staging environment to get an understanding what new quickstarts are: https://stage.code.quarkus.io/[https://stage.code.quarkus.io/]
```
Comment (ia3andy) [inline independent-projects/tools/codestarts/README.adoc:11]: You can point to code.quarkus.io directly in the doc, I will promote staging to production later today..

Comment (ia3andy) [inline independent-projects/tools/codestarts/README.adoc:11]:
```suggestion
Try generate a project with example code to get an understanding what new quickstarts are: https://code.quarkus.io/[https://code.quarkus.io/]
```

File: devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/examples/resteasy-jackson-example/java/src/main/java/org/acme/rest/json/Fruit.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package org.acme.rest.json;
```
Comment (ia3andy) [inline devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/examples/resteasy-jackson-example/java/src/main/java/org/acme/rest/json/Fruit.java:1]: I am not sure but maybe better to use `org.acme.rest.resteasyjackson` as package name?

Comment (jantosi) [reply devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/examples/resteasy-jackson-example/java/src/main/java/org/acme/rest/json/Fruit.java:1]: Good suggestion! I'll change it.

**Comment by jantosi** [comment]
Pushed a simple change to trigger the build again - JVM core-dumped:
https://github.com/quarkusio/quarkus/runs/1234266259?check_suite_focus=true#step:6:70362

**Review by ia3andy — APPROVED** [review]
Nice!

**Comment by ia3andy** [comment]
In another PR, we will need to update the guide: https://github.com/quarkusio/quarkus/issues/12642

**Pull request merged**